hibernate-reactive-core/src/main/java/org/hibernate/reactive/id/insert/ReactiveInsertReturningDelegate.java

```diff
@@ -7,51 +7,42 @@
 
 import java.lang.invoke.MethodHandles;
 import java.sql.PreparedStatement;
-import java.util.ArrayList;
 import java.util.List;
-import java.util.concurrent.CompletionStage;
 
 import org.hibernate.dialect.Dialect;
 import org.hibernate.dialect.MySQLDialect;
+import org.hibernate.dialect.SQLServerDialect;
 import org.hibernate.engine.jdbc.mutation.JdbcValueBindings;
 import org.hibernate.engine.jdbc.mutation.group.PreparedStatementDetails;
 import org.hibernate.engine.spi.SessionFactoryImplementor;
 import org.hibernate.engine.spi.SharedSessionContractImplementor;
 import org.hibernate.generator.EventType;
-import org.hibernate.generator.values.GeneratedValueBasicResultBuilder;
 import org.hibernate.generator.values.GeneratedValues;
 import org.hibernate.generator.values.internal.TableUpdateReturningBuilder;
-import org.hibernate.id.insert.AbstractReturningDelegate;
 import org.hibernate.id.insert.InsertReturningDelegate;
 import org.hibernate.id.insert.TableInsertReturningBuilder;
 import org.hibernate.jdbc.Expectation;
 import org.hibernate.metamodel.mapping.BasicEntityIdentifierMapping;
 import org.hibernate.persister.entity.EntityPersister;
 import org.hibernate.reactive.logging.impl.Log;
 import org.hibernate.reactive.logging.impl.LoggerFactory;
-import org.hibernate.reactive.session.ReactiveConnectionSupplier;
-import org.hibernate.sql.ast.tree.expression.ColumnReference;
 import org.hibernate.sql.model.ast.MutatingTableReference;
 import org.hibernate.sql.model.ast.builder.TableMutationBuilder;
 
 import static java.sql.Statement.NO_GENERATED_KEYS;
 import static org.hibernate.generator.EventType.INSERT;
-import static org.hibernate.generator.values.internal.GeneratedValuesHelper.getActualGeneratedModelPart;
-import static org.hibernate.reactive.generator.values.internal.ReactiveGeneratedValuesHelper.getGeneratedValues;
 
 /**
  * @see InsertReturningDelegate
  */
-public class ReactiveInsertReturningDelegate extends AbstractReturningDelegate implements ReactiveAbstractReturningDelegate {
+public class ReactiveInsertReturningDelegate extends ReactiveAbstractReturningDelegate {
 
 	private static final Log LOG = LoggerFactory.make( Log.class, MethodHandles.lookup() );
 
-	private final EntityPersister persister;
 	private final MutatingTableReference tableReference;
-	private final List<ColumnReference> generatedColumns;
 
 	public ReactiveInsertReturningDelegate(EntityPersister persister, EventType timing) {
-		this( persister, timing, false );
+		this( persister, timing, true );
```

Comment:
In this class there's an extra constructor that set the value to false for MySQL (see method [supportsArbitraryValues](https://github.com/hibernate/hibernate-reactive/blob/23b63a692e802d67ac177737d941e150d3f44364/hibernate-reactive-core/src/main/java/org/hibernate/reactive/id/insert/ReactiveInsertReturningDelegate.java#L61)).

I think we can remove it, MySQL works fine now (probably because of the changes in this PR).